gtars-overlaprs/src/genome_index.rs

```diff
@@ -0,0 +1,678 @@
+//! Genome-wide interval indexing for efficient multi-chromosome overlap queries.
+//!
+//! This module provides [`GenomeIndex`](crate::genome_index::GenomeIndex), a data structure that extends single-chromosome
+//! overlap data structures (like [`AiList`](crate::AiList) or [`Bits`](crate::Bits)) to handle genome-wide queries
+//! across multiple chromosomes efficiently.
+//!
+//! # Overview
+//!
+//! While [`AiList`](crate::AiList) and [`Bits`](crate::Bits) are designed for queries on a single chromosome,
+//! [`GenomeIndex`](crate::genome_index::GenomeIndex) manages a collection of these structures - one per chromosome - enabling
+//! efficient queries across the entire genome.
+//!
+//! # Examples
+//!
+//! ## Basic Usage
+//!
+//! ```
+//! use gtars_overlaprs::{OverlapperType, genome_index::IntoGenomeIndex};
+//! use gtars_core::models::{Region, RegionSet};
+//!
+//! // Create a genome-wide set of intervals (e.g., gene annotations)
+//! let genes = vec![
+//!     Region { chr: "chr1".to_string(), start: 1000, end: 2000, rest: Some("BRCA1".to_string()) },
```

Comment:
I think using a Region object like this is really atypical. It's also going to be slower.

It feels better to have your examples use vectors of regions, rather than individual region elements.